### PR TITLE
UtilityToken.setup method

### LINKED ISSUE #267: Update utility token contract
Update Utility token contract (2)
- Make it master copy non upgradable 
- update setup function
    
```
    function setup(
        string calldata _symbol,
        string calldata _name,
        uint8 _decimals,
        uint256 _totalTokenSupply,
        address _erc20Cogateway,
        address _valueToken
    )
        external
```

### PR DESCRIPTION
**Pull request by 0xsarvesh**
⚠️  Blocked by #266 

fixes #267 

1. Made utility token setup method external
2. Added value token param
3. Removed unused event
4. Utility token is now MasterCopyNonUpgradable

### CODE REVIEW

File: contracts/utility-token/UtilityToken.sol [hunk 1]
```diff
@@ -51,6 +49,50 @@ contract UtilityToken is ERC20Token {
 
     /* External Functions */
 
+    /**
```
Comment (pgev) [inline contracts/utility-token/UtilityToken.sol:52]: Please, define `\pre` and `\post` documentation tags for this external function.

File: contracts/utility-token/UtilityToken.sol [hunk 2]
```diff
@@ -51,6 +49,50 @@ contract UtilityToken is ERC20Token {
...
+        uint8 _decimals,
+        uint256 _totalTokenSupply,
+        address _consensusCogateway,
+        address _valueToken
+    )
+        external
+    {
+        require(
+            consensusCogateway == address(0),
+            "ConsensusCogateway address is already set."
```
Comment (pgev) [inline contracts/utility-token/UtilityToken.sol]: I am assuming we do this check to make sure contract is initialized/setup only once. Hence, I would propose to update the require message to `Contract has been already setup`.

**Review by pgev — DISMISSED** [review]
@sarvesh-ost thanks for changes. Please, see my only documentation comments below.

File: contracts/utility-token/UtilityToken.sol (hunk 1 above)
Comment (0xsarvesh) [reply contracts/utility-token/UtilityToken.sol:52]: Done

File: contracts/utility-token/UtilityToken.sol (hunk 2 above)
Comment (0xsarvesh) [reply contracts/utility-token/UtilityToken.sol]: Done

File: contracts/utility-token/UtilityToken.sol [hunk 3]
```diff
@@ -51,6 +49,63 @@ contract UtilityToken is ERC20Token {
...
+     *         and the consensusCogateway address.
+     *
+     * @param _symbol Symbol of token.
+     * @param _name Name of token.
+     * @param _decimals Decimal of token.
+     * @param _totalTokenSupply Total token supply.
+     * @param _consensusCogateway ConsensusCogateway contract address.
+     * @param _valueToken Address of value token contract.
+     *
+     * \pre Setup is not already called.
```
Comment (deepesh-kn) [inline contracts/utility-token/UtilityToken.sol]: ` * \pre This function can be called only once.`
or
` * \pre Setup function can be called only once.`

**Review by deepesh-kn — COMMENTED** [review]
Looks good. Just one comment.
One more change  I will request you to remove ` emit Transfer(address(0), _account, _value);` from mint function. This change is not part of the ticket but I think that its just removal of one line

File: contracts/utility-token/UtilityToken.sol (hunk 3 above)
Comment (0xsarvesh) [reply contracts/utility-token/UtilityToken.sol]: Done

**Review by deepesh-kn — APPROVED** [review]

**Pull request merged**